mino/minogrpc/server.go

```diff
@@ -530,28 +544,33 @@ func (s *sender) sendSingle(msg proto.Message, from, to mino.Address) error {
 		routingTo = s.address
 	}
 
-	player, participantFound := s.participants[routingTo.String()]
+	logTo := routingTo
+	// This is the case we are responding to our stream, which replies to the
+	// orchestrator that created the stream.
+	if routingTo == s.address {
+		logTo = s.rootAddr
+	}
+
+	playerItf, participantFound := s.participants.Load(routingTo.String())
 	if !participantFound {
 		return xerrors.Errorf("failed to send a message to my child '%s', "+
 			"participant not found in '%v'", routingTo, s.participants)
 	}
 
-	envelope := &Envelope{
-		From:    from.String(),
-		To:      []string{to.String()},
-		Message: msgAny,
-	}
-	envelopeAny, err := s.encoder.MarshalAny(envelope)
-	if err != nil {
-		return xerrors.Errorf("couldn't marshal envelope: %v", err)
+	player, ok := playerItf.(overlayStream)
+	if !ok {
+		fabric.Logger.Fatal().Msg("not ok")
```

Comment:
I can't believe I'm the one saying an error message is too short ! But it should slightly more detailed. Why would that happen ?

Reply:
Ugh, you're so right! I forgot to update this one.... : /